### PR TITLE
Allow CURL constant names

### LINKED ISSUE #71: Add support for string curl client params
Maybe we could accept curl client constants as well. Before passing to the client, we could resolve the constant names to actual values. What do you think?

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #71

### CODE REVIEW

File: ClientFactory/CurlFactory.php [hunk 1]
```diff
@@ -40,6 +40,18 @@ public function createClient(array $config = [])
             throw new \LogicException('To use the Curl client you need to install the "php-http/curl-client" package.');
         }
 
+        // Try to resolve curl constant names
+        foreach ($config as $key => $value) {
+            // If the value starts with 'CURL' we assume it is a reference to a constant.
+            if (strpos($key, 'CURL') === 0) {
+                $contantValue = constant($key);
```
Comment (dbu) [inline ClientFactory/CurlFactory.php]: is there something to check if a constant exists?

and s/contantValue/constantValue ;-)

Comment (Nyholm) [inline ClientFactory/CurlFactory.php]: If the constant does not exist constant() will return null. 

Question: is it true that the only config parameters to the curl client are integers or constants named "CURL*"? If so, we should throw a exception.

Comment (dbu) [inline ClientFactory/CurlFactory.php]: according to http://php.net/manual/en/function.curl-setopt.php it must be int. so either it already is an int here, or its a constant. maybe we could do

```
if (!is_int($key)) {
    $constantValue = constant($key);
    if (null === $constantValue) {
        throw... $key is not an integer or a CURL constant.
    }
    unset...,
    $config[$constantValue] = $value;
}
```

Comment (dbu) [inline ClientFactory/CurlFactory.php]: not sure if we should force the constants to start with CURL - maybe there could be a reason to have some other constant that does not start with CURL.

is `constant()` case sensitive?

Comment (sagikazarmark) [inline ClientFactory/CurlFactory.php]: Invalid constants (integers) will end up in an error upon setopt anyway.

Comment (Nyholm) [inline ClientFactory/CurlFactory.php]: Okey. Then I'm safe to assume that everything that comes in is either an Int or a name of a constant. If the constant is not defined I'll throw an exception. If the Int is invalid CURL will throw an exception

**Comment by Nyholm** [comment]
Thank you for the feedback. I've updated the PR accordingly.

**Comment by dbu** [comment]
awesome, thanks!

**Comment by Nyholm** [comment]
Thank you for merging.

**Pull request merged**